pkg/shell/cockpit-server.js

```diff
@@ -188,14 +188,32 @@ PageServer.prototype = {
     enter: function() {
         var self = this;
 
-        /* TODO: Need to migrate away from old dbus */
-        self.client = shell.dbus(null);
-
-        self.manager = self.client.get("/com/redhat/Cockpit/Manager",
-                                       "com.redhat.Cockpit.Manager");
-        $(self.manager).on('AvatarChanged.server', $.proxy (this, "update_avatar"));
+        self.client = cockpit.dbus('org.freedesktop.hostname1');
+        self.hostname_proxy = self.client.proxy('org.freedesktop.hostname1',
+                                     '/org/freedesktop/hostname1',
+                                     {'watch' : true});
```

Comment:
I think 'watch' is true by default. You have to set it to false to turn off watching.